Update passing of initial props on android

mchowning opened the pull request:
Please see the related [WPAndroid PR](https://github.com/wordpress-mobile/WordPress-Android/pull/12444) for information about this PR.

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [X] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [X] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [X] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [X] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

mchowning committed:
Use GutenbergProps object for initialProps

mchowning committed:
Simplify GutenbergProps object

cameronvoell reviewed (changes requested):
Changes are awesome. However, looks like we still need to update [MainActivity.java](https://github.com/WordPress/gutenberg/blob/fbf517d141e0373a2e6feb92c8c0e4d0a9f58dca/packages/react-native-editor/android/app/src/main/java/com/gutenberg/MainActivity.java#L31-L33) in the Demo App to fix the Android Demo app and CI.

mchowning committed:
Update demo app with GutenbergProps changes

mchowning commented:
> looks like we still need to update MainActivity.java in the Demo App to fix the Android Demo app and CI

Right you are @cameronvoell ! 👍 I've updated this PR to fix the demo app in daefab8 and updated the submodule refs in the gb-mobile and WPAndroid PRs.
